### ISSUE TITLE
[Bug]: Unhelpful error message when missing @added decorator

### ISSUE THREAD

**Issue by mario-guerra** [microsoft/typespec#3920]
### Describe the bug

I'm revising the pet store example and encountered a cryptic error when I forgot to consistently  use the '@added' decorator to specify versioning in operations added for the toy resource. 

This code uses two instances of the `@added` decorator:

```
   // Operations under the 'toys' namespace are only available in v2 of the API
    @route("{petId}/toys/{toyId}")
    namespace Toys {
        @added(Versions.v2)
        @get
        op getToy(@path petId: int32, @path toyId: int32): OkResponse & {
            @body toy: Toy;
        } | NotFoundResponse & {
            @body error: NotFoundError;
        };
        
        @added(Versions.v2)
        @useAuth(BearerAuth)
        @put
        op updateToy(@path petId: int32, @path toyId: int32, @body toy: Toy): OkResponse & {
            @body updatedToy: Toy;
        } | NotFoundResponse & {
            @body error: NotFoundError;
        };
    }
```

If I remove the second `@added` decorator, I get the following error, which doesn't help me determine what the issue is:
`error
@typespec/openapi3/invalid-schema
Couldn't get schema for type never
<unknown location>:1:1`

Can the error checking be improved to correctly identify the cause?

### Reproduction

[playground link to error case](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZyI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwO9AVUmVzdNEVVslHOwoKLyoqCiAqIFRoaXMgaXMgYSBzYW1wbGUgUGV0IFN0b3JlIEFQSS4KICovCkBzZXJ2aWNlKHsKICAgIHRpdGxlOiAiySoiLAp9KcYmZXIoIuQA4HM6Ly9leMVSLmNvbSIsICJTaW5nbGUgxiYgZW5kcG9pbnQiKQpA5wDfZWQo5wCicykKbmFtZXNwYWPlAJHFZjsKZW51bSDIIyDmAI12MTogIjEuMCIsxg8yOiAiMsUPfekA5OQAqW1vZGVsLCDkAU9pZnnkARdhIHBldCDEaCwgYWfEBW5kIGtpbmTmAPXFMcU7xmlpZDogaW50MzI7xQ%2FFBUBtaW5MZW5ndGgoMSnFEsRPOiBzdHLmAWXIJVZhbHVlKDDGJEBtYXjGETEwxxNhZ2XJXsUR5ACIOiAiZG9nIiB8ICJjYXTFCGZpc2jFCWJpcmTFCXJlcHRpbGUiO%2BsA8FRvefUA8HRveeUAoecB1mFkZOsBcy52MinnAPXEQPUA9e4A3spxRXJyb3LGc3PHXWXEE8dQVmFsaWRhdGlvbsYpxlxjb2TkAkZWQUxJREFUSU9OX0VSUk9SIsZrbWVzc%2BUBFshuxBVkZXRhaWxzyBVbXeQAhc1rTm90Rm91bmTTaU5PVF9GT1VORNhiz0tJbnRlcm5hbFPlArHTUUlOVEVSTkFMX1NFUlZFUv4Av%2BoBLUN1c3RvbSByZXNwb25zZeYBNyBmb3IgNTAwyXYgxnfmAVvsApLzAJdSyEvmAJ9Ac3RhdHVzQ29kZSDKCzrEX%2BYAnUBib2R5IOUA7DrUUO0Arm9tbW9uIHBhcmFtZXRlcnPnALB1c2VkIGFjcm9zcyBtdWx0aeQEPG9wZXLlAejnAgbGK8ZDUMpC5wCncXVlcnnFC3NraXA%2F7gMByx50b9EdaGVhZGVyxQxyZXF1ZXN0SUT0AWzoBN50aGUgcm9v5gPv5gRX5AFxxBfxBPAgQWxs6wDKIGFyZSB1bmRlxi8vcGV0cyBwYXRo5wLpcm91dGUoIsUYIu8EuOkA62dldMUJb3AgbGlzdMQbKC4uLvABGik6IE9r6QHOJsc%2B6gGzxGQ6xFnkAvrEF33oARHLYWdlxGAoQOQApMQySegDxCzoATggaWZNYXRjaD%2FoATPKGvEBTvsAk8czRVRhZ%2B4Dp%2B0AseUAsMYYfSB86QOh1lPsApntA8vHQTsK5gDwLy8gT%2BoBouoBnid0b3lzJ%2BsB5uQBvW9ubHkgYXZhaWxhYmxlIGluIHYyIG9mxTFBUEnFUegBx3vlATF9L8RIL3t0b3lJZH0i6gXbxlVUb3noAdrFPPMFSckc6AGZ6gGdVG959gGdxRTFc8cU%2BgF86gEtdG95OuQAkeoBev8BZv8BaucBasVJygvJCUB1c2VBdXRoKEJlYXJlcsQL6wDtcHXtAO11cGRhdGX%2FAPDrAPAs7wDU%2FwEAxSzGZGRU%2FwEH3UT%2FAQfGC30KfSAg&e=%40typespec%2Fopenapi3&options=%7B%22emit%22%3A%5B%22%40typespec%2Fopenapi3%22%5D%7D)

[playground link to working case](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZyI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwO9AVUmVzdNEVVslHOwoKLyoqCiAqIFRoaXMgaXMgYSBzYW1wbGUgUGV0IFN0b3JlIEFQSS4KICovCkBzZXJ2aWNlKHsKICAgIHRpdGxlOiAiySoiLAp9KcYmZXIoIuQA4HM6Ly9leMVSLmNvbSIsICJTaW5nbGUgxiYgZW5kcG9pbnQiKQpA5wDfZWQo5wCicykKbmFtZXNwYWPlAJHFZjsKZW51bSDIIyDmAI12MTogIjEuMCIsxg8yOiAiMsUPfekA5OQAqW1vZGVsLCDkAU9pZnnkARdhIHBldCDEaCwgYWfEBW5kIGtpbmTmAPXFMcU7xmlpZDogaW50MzI7xQ%2FFBUBtaW5MZW5ndGgoMSnFEsRPOiBzdHLmAWXIJVZhbHVlKDDGJEBtYXjGETEwxxNhZ2XJXsUR5ACIOiAiZG9nIiB8ICJjYXTFCGZpc2jFCWJpcmTFCXJlcHRpbGUiO%2BsA8FRvefUA8HRveeUAoecB1mFkZOsBcy52MinnAPXEQPUA9e4A3spxRXJyb3LGc3PHXWXEE8dQVmFsaWRhdGlvbsYpxlxjb2TkAkZWQUxJREFUSU9OX0VSUk9SIsZrbWVzc%2BUBFshuxBVkZXRhaWxzyBVbXeQAhc1rTm90Rm91bmTTaU5PVF9GT1VORNhiz0tJbnRlcm5hbFPlArHTUUlOVEVSTkFMX1NFUlZFUv4Av%2BoBLUN1c3RvbSByZXNwb25zZeYBNyBmb3IgNTAwyXYgxnfmAVvsApLzAJdSyEvmAJ9Ac3RhdHVzQ29kZSDKCzrEX%2BYAnUBib2R5IOUA7DrUUO0Arm9tbW9uIHBhcmFtZXRlcnPnALB1c2VkIGFjcm9zcyBtdWx0aeQEPG9wZXLlAejnAgbGK8ZDUMpC5wCncXVlcnnFC3NraXA%2F7gMByx50b9EdaGVhZGVyxQxyZXF1ZXN0SUT0AWzoBN50aGUgcm9v5gPv5gRX5AFxxBfxBPAgQWxs6wDKIGFyZSB1bmRlxi8vcGV0cyBwYXRo5wLpcm91dGUoIsUYIu8EuOkA62dldMUJb3AgbGlzdMQbKC4uLvABGik6IE9r6QHOJsc%2B6gGzxGQ6xFnkAvrEF33oARHLYWdlxGAoQOQApMQySegDxCzoATggaWZNYXRjaD%2FoATPKGvEBTvsAk8czRVRhZ%2B4Dp%2B0AseUAsMYYfSB86QOh1lPsApntA8vHQTsK5gDwLy8gT%2BoBouoBnid0b3lzJ%2BsB5uQBvW9ubHkgYXZhaWxhYmxlIGluIHYyIG9mxTFBUEnFUegBx3vlATF9L8RIL3t0b3lJZH0i6gXbxlVUb3noAdrFPPMFSckc6AGZ6gGdVG959gGdxRTFc8cU%2BgF86gEtdG95OuQAkeoBev8BZv8BaucBasVJygv9AON1c2VBdXRoKEJlYXJlcsQLyx1wde0BAHVwZGF0Zf8BA%2BsBAyzvAOf%2FARPFLMZkZFT%2FARrdRP8BGsYLfQp9ICA%3D&e=%40typespec%2Fopenapi3&options=%7B%22emit%22%3A%5B%22%40typespec%2Fopenapi3%22%5D%7D)

### Checklist

- [X] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [X] Check that there isn't already an issue that request the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/Microsoft/typespec/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by timotheeguerin** [comment]
[Smaller repro](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZyI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwO9AVUmVzdNEVVslHOwoKLyoqCiAqIFRoaXMgaXMgYSBzYW1wbGUgUGV0IFN0b3JlIEFQSS4KICovCkBzZXJ2aWNlCkDnAIllZCjHTHMpCm5hbWVzcGFjxTvFOjsKZW51bSDIIyB7CiAgdjE6ICIxLjAiLMQNMjogIjLFDX3qAIpveSBtb2RlbCwg5AD1aWZ55AC9YSB0b3kgxGTnAIxhZGTLfy52MikKxTbFQMRqaWQ6IGludDMyOwogxTk6IHN0cuUA9X0KCkB1c2VBdXRoKEJlYXJlcsQLKQpAcHV0Cm9wIHVwZGF0ZVRveShAYm9kecR7OsRYKTogdm9pZDsK&e=%40typespec%2Fopenapi3&options=%7B%22emit%22%3A%5B%22%40typespec%2Fopenapi3%22%5D%7D) Seems to be we are not detecting operation parameters are being a problem if they use a type that is unavailable

**Comment by mario-guerra** [comment]
Thanks @timotheeguerin. Is the error due to the use of the 'Toy' model which is defined as 'v2' in an op that is implicitly defined as 'v1'?

**Comment by timotheeguerin** [comment]
yeah, this is a recurring issue we have with versioning there is always other cases where we don't detect that and cause those `never` values

**Comment by mario-guerra** [comment]
Is it common enough to have a FAQ entry about in our docs?

**Comment by timotheeguerin** [comment]
Yeah whenever we fix an issue a little later someone finds a new case where it still happens so could be good to have this in the faq, just very annoying to diagnose

**Comment by mario-guerra** [comment]
Ok I'll create a FAQ entry for this. Is this issue always related to improper application of versioning?

**Comment by timotheeguerin** [comment]
Seeing the `never` type yeah pretty much(unless user actually did use never in that spot but then it should be quite clear to them)
